EKF/control.cpp

```diff
@@ -667,23 +683,21 @@ void Ekf::controlGpsFusion()
 				_time_last_on_ground_us = _time_last_imu;
 			}
 			const bool recent_takeoff = _control_status.flags.in_air && !isTimedOut(_time_last_on_ground_us, 30000000);
-			const bool reset_yaw_to_EKFGSF = (do_reset || _do_emergency_yaw_reset || stopped_following_gps_velocity) &&
+			const bool do_yaw_vel_pos_reset = (do_vel_pos_reset || _do_ekfgsf_yaw_reset || stopped_following_gps_velocity) &&
 								recent_takeoff &&
-								isTimedOut(_emergency_yaw_reset_time, 5000000);
+								isTimedOut(_ekfgsf_yaw_reset_time, 5000000);
 
-			if (reset_yaw_to_EKFGSF) {
-				if (resetYawToEKFGSF()) {
-					_emergency_yaw_reset_time = _time_last_imu;
-					_do_emergency_yaw_reset = false;
+			if (do_yaw_vel_pos_reset && resetYawToEKFGSF()) {
```

Comment:
Wait, this is not correct, it means that we constantly reset to the yaw estimator when using GPS

Reply:
I think this is OK in current code This reset can be triggered two ways, one is via timeouts and the other is via _do_emergency_yaw_reset. Both timeouts and _do_emergency_yaw_reset are cleared and also the next reset is blocked by the isTimedOut(_emergency_yaw_reset_time, 5000000) condition that must be met.